File: 程序员面试金典第6版/面试题 01.04. Palindrome Permutation LCCI.py
class Solution:
    def canPermutePalindrome(self, s: str) -> bool:
        l = len(s)
        if l == 1 or len(set(s)) == 1:
            return True

        have = set()

        for i in range(l):
            if s[i] in have:
                have.remove(s[i])
            else:
                have.add(s[i])

        return len(have) <= 1

File: 程序员面试金典第6版/面试题 01.04. Palindrome Permutation LCCI.java
class Solution {
    public boolean canPermutePalindrome(String s) {
        if(s == null) return false;
        if(s.length() == 1) return true;

        char[] chs = s.toCharArray();
        Set<Character> set = new HashSet();
        for(char c: chs) {
            if(set.contains(c)) {
                set.remove(c);
            } else {
                set.add(c);
            }
        }
        return set.size() <= 1;
    }
}